packages/client/src/Components/Mapping/index.jsx

```diff
@@ -130,32 +128,74 @@ const Mapping = ({ currentTurn, role, currentWargame, selectedForce, allForces,
     sendMessage(FORCE_LAYDOWN, param)
   }
 
+  const visChangesFunc = changes => {
+    sendMessage(VISIBILIY_CHANGES, changes)
+
+    /** note: we aren't currently receiving messages that we've sent. So we
+     * to trigger the reducer ourselves
+     */
+    handleVisibilityChanges(changes, allForces)
+  }
+
   /** callback to tell UI that we've got control of a platform in this UI */
   const declareControlOf = (force, name, platformType) => {
-    console.log('User can control:', name)
   }
 
   const routeComplete = (/* string */force, /* string */asset, /* object */payload) => {
     console.log('Planned Route:', force, asset, payload)
   }
 
   const clearControlledAssets = () => {
-    console.log('Clearing list of controlled assets')
   }
 
   const formRequestCallback = (form, payload) => {
     console.log('Popup form requested for:', form, payload)
-    setShowForm(true)
-    setFormPos(payload.screenPos)
   }
 
-  useEffect(() => {
-    // double-check where we are
-    console.log(new Date(), 'TURN:', phaseRef.current, currentTurnRef.current)
+  const createThisMarker = (asset, grid, force) => {
+    // set the asset location
+    asset.loc = grid.hexNamed(asset.position).centrePos
+
+    const inForceLaydown = hasPendingForces(forcesRef.current, myForceRef.current)
+
+    // set the asset force
+    asset.force = force.name
+
+    var assetIsDraggable
+    switch (phase) {
+      case 'adjudication':
+        assetIsDraggable = ((myForceRef.current === 'umpire') || (inForceLaydown && (asset.force === myForceRef.current)))
+        break
+      case 'planning':
+        assetIsDraggable = myForceRef.current !== 'umpire' && asset.force === myForceRef.current
+        break
+      default:
+        console.log('Error - unexpected game phase encountered in Mapping component')
+    }
+
+    const userIsUmpire = myForceRef.current === 'umpire'
+    const marker = markerFor(asset, force.name, myForceRef.current, platformTypesRef.current, assetIsDraggable, userIsUmpire)
+
+    // did we create one?
```

Comment:
This is another example where an imperative comment with context would be more useful. Who is `we` and what did we possible create `one` of?